File: scripts/mongodb/queries/email-sha1s.js
// Finds all email addresses of users, normalizes, and produces SHA1 hashes.
// Run while piping output to user_emails.txt.
// Then run python email-sha1s.py to get user_sha1s.txt.

var normalizedEmails = [];
var usersWithEmails = db.users.find({emailLower: {$exists: true}}, {emailLower: 1}).forEach(function(u) {
  if(u.emailLower && u.emailLower.trim().length)
    normalizedEmails.push(u.emailLower.trim().toLowerCase().replace('googlemail', 'gmail').replace(/\.(?=.*@)/g, '').replace(/\+.*@/g, '@'));
});
normalizedEmails.forEach(function(e) { print(e); });

File: scripts/mongodb/queries/email-sha1s.py
from __future__ import with_statement
import hashlib

with open("user_emails.txt", "r") as f:
  emails = f.read().strip().split('\n')

sha1s = [hashlib.sha1(e).hexdigest() for e in emails]

with open("user_sha1s.txt", "w") as f:
  f.write("\n".join(sha1s) + "\n")
